src/templates/index.jsx

```diff
@@ -453,13 +461,23 @@ const Layout = React.createClass({
         this.noteListVisibilityUpdater = updater;
     },
 
+    storeDetailViewRef(ref) {
+        this.detailView = ref;
+    },
+
     render() {
+        const { previouslySelectedNote, selectedNote } = this.state;
+
         const currentNote = find(
             this.props.notes,
             matchesProperty('id', this.props.selectedNoteId)
         );
         const filteredNotes = this.filterController.getFilteredNotes(this.props.notes);
 
+        if (this.detailView && selectedNote !== previouslySelectedNote) {
+            this.detailView.scrollTop = 0;
```

Comment:
hm. is it common to adjust scroll in render? we might re-render but not want to scroll for some reason, such as when a new note appears from the server. would this be more appropriate somewhere like `componentWillUpdate` or `componentDidUpdate`?